pandas/core/frame.py

```diff
@@ -730,6 +730,7 @@ def __init__(
                     columns,
                     dtype=dtype,
                     typ=manager,
+                    copy=copy,
```

Comment:
This is necessary to avoid a copy when transforming from series to df in mixed case. Should we hide this behind the CoW flag?

Reply:
I suppose this will change behaviour (for the default non-CoW mode) if you pass `copy=False` ? (and so actually for the default copy=None, as that is translated to copy=False in such a case). Currently that still does make a copy always:

```
In [9]: s = pd.Series([1, 2, 3], name="a")

In [10]: df = pd.DataFrame(s, copy=False)

In [11]: s.iloc[0] = 100

In [12]: df
Out[12]: 
   a
0  1
1  2
2  3
```

One could consider that a bug (but then we also need to add testes for that case here), but given that we probably want to change that in the future to actually behave as a copy (i.e. the current behaviour), I would maybe not fix the "bug".
So, yes, then I would hide this behind a `using_copy_on_write()` check.

Reply:
Added the option check